client/question/components/LearnerViewDirective.js

```diff
@@ -925,6 +1037,69 @@ tie.directive('learnerView', [function() {
           $scope.scrollToTopOfFeedbackWindow();
         };
 
+        /**
+         * Sets the code in the code editor to the snapshot index
+         * passed in as a parameter.
+         *
+         * @param {number} selectedSnapshotIndex The snapshot index
+         * selected in the previous snapshots dropdown.
+         */
+        $scope.revertToSelectedSnapshot = function(selectedSnapshotIndex) {
+          $scope.currentSnapshotIndex = selectedSnapshotIndex;
+          var selectedSnapshot = null;
+          if (selectedSnapshotIndex === 0) {
+            selectedSnapshot = SessionHistoryService.getStarterCodeSnapshot();
+            $scope.previousButtonIsDisabled = true;
+          } else {
+            selectedSnapshot = SessionHistoryService.getPreviousSnapshot(
+              selectedSnapshotIndex);
+            $scope.previousButtonIsDisabled = false;
+          }
+          if (selectedSnapshot === null) {
+            throw Error('Could not retrieve code for snapshot at index ' +
+              selectedSnapshotIndex);
+          } else {
+            $scope.editorContents.code = selectedSnapshot;
+          }
+          $scope.snapshotMenuIsOpen = false;
+        };
+
+        /**
+         * Sets the code in the code editor to the previous snapshot.
+         */
+        $scope.revertToPreviousSnapshot = function() {
+          var previousSnapshotIndex = $scope.currentSnapshotIndex - 1;
+          $scope.currentSnapshotIndex = previousSnapshotIndex;
+          var previousSnapshot = null;
+          if (previousSnapshotIndex >= 0) {
+            if (previousSnapshotIndex === 0) {
+              previousSnapshot = SessionHistoryService.getStarterCodeSnapshot();
+              $scope.previousButtonIsDisabled = true;
+            } else {
+              previousSnapshot = SessionHistoryService.getPreviousSnapshot(
+                previousSnapshotIndex);
+              $scope.previousButtonIsDisabled = false;
+            }
+          }
+          if (previousSnapshot === null) {
+            throw Error('Could not retrieve code for snapshot at index ' +
+              previousSnapshotIndex);
+          } else {
+            $scope.editorContents.code = previousSnapshot;
+          }
+        };
+
+        /**
+         * Open and close the snapshot menu.
+         */
+        $scope.showSnapshotMenu = function() {
+          if ($scope.snapshotMenuIsOpen === true) {
```

Comment:
if ($scope.snapshotMenuIsOpen) { should be sufficient, unless you're assuming it can be anything beyond true or false.